src/audio/copier/copier_gain.h

```diff
@@ -0,0 +1,136 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2024 Intel Corporation. All rights reserved.
+ *
+ * Author: Ievgen Ganakov <ievgen.ganakov@intel.com>
+ */
+
+#ifndef __SOF_COPIER_GAIN_H__
+#define __SOF_COPIER_GAIN_H__
+
+#include <sof/audio/buffer.h>
+#include <xtensa/tie/xt_hifi3.h>
+#include <ipc4/base_fw.h>
+#include <ipc/dai.h>
+
+/**
+ * @file copier_gain.h
+ * @brief Header file containing definitions and functions related to audio gain
+ * processing for a copier module.
+ *
+ * This file provides functions, constants and structure definitions for applying gain to
+ * input audio buffers, both in 16-bit and 32-bit container formats. The gain can be
+ * applied in different directions (addition or subtraction) and has three modes:
+ * - static gain
+ * - transition gain (fade-in/fade-out)
+ * - transition mute.
```

Comment:
this is very unclear. a mute is typically when a fade-in/fade-out is done. So what's the difference ?